mdata/cache/chunk_cache.go

```diff
@@ -0,0 +1,96 @@
+package cache
+
+import (
+	"flag"
+	"sync"
+
+	"github.com/raintank/metrictank/iter"
+	"github.com/rakyll/globalconf"
+)
+
+var (
+	maxSize uint64
+)
+
+func ConfigSetup() {
+	flags := flag.NewFlagSet("chunk-cache", flag.ExitOnError)
+	// (1024 ^ 3) * 4 = 4294967296
+	flags.Uint64Var(&maxSize, "max-size", 4294967296, "Maximum size of chunk cache in bytes")
+	globalconf.Register("chunk-cache", flags)
+}
+
+type CacheChunk struct {
+	Ts      uint32
+	Next    uint32
+	Prev    uint32
+	itergen iter.IterGen
+}
+
+type MetricCache struct {
```

Comment:
I agree, I'm just not sure what to rename it to. I think the name `ChunkCache` for the most outer scoped struct still makes sense, because it is theoretically possible to have more than one of them (I wouldn't know why).
How about I rename `MetricCache` to `ChunkCacheMetric`? I know you don't like long names, but I think it's kind of logical because that makes it clear that this is a more specifically scoped piece that belongs to `ChunkCache`. To make the names shorter I could also just rename `ChunkCache -> CCache` & `ChunkCacheMetric -> CCacheMetric`.